### PR TITLE
Migrate storage to dom

### PR DESCRIPTION
**Pull request by armanbilge**
Towards #545.

### CODE REVIEW

File: src/main/scala/org/scalajs/dom/experimental/storage/package.scala [hunk 1]
```diff
@@ -1,30 +1,22 @@
...
 import scala.language.implicitConversions
-import scala.scalajs.js
 
 /** https://storage.spec.whatwg.org/ */
 package object storage {
 
-  implicit def toNavigatorStorage(navigator: dom.Navigator): NavigatorStorage =
-    navigator.asInstanceOf[NavigatorStorage]
+  @deprecated("use dom.Navigator directly instead", "2.0.0")
+  implicit def toNavigatorStorage(navigator: dom.Navigator): dom.Navigator =
```
Comment (japgolly) [inline src/main/scala/org/scalajs/dom/experimental/storage/package.scala:10]: This is `identity. Either it shouldn't exist at all, or if it's for compat, it shouldn't be implicit.

File: src/main/scala/org/scalajs/dom/storage.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+@js.native
+trait StorageManager extends js.Any {
+  def persisted(): js.Promise[Boolean] = js.native
+  def persist(): js.Promise[Boolean] = js.native
+  def estimate(): js.Promise[StorageEstimate] = js.native
+}
+
+@js.native
+trait StorageEstimate extends js.Any {
```
Comment (japgolly) [inline src/main/scala/org/scalajs/dom/storage.scala:13]: This (and above) should extends `js.Object`, not `js.Any`

File: src/main/scala/org/scalajs/dom/lib.scala [hunk 3]
```diff
@@ -325,6 +325,9 @@ class Navigator
     *   The data parameter is the ArrayBufferView, Blob, DOMString, or FormData data that is to be transmitted.
     */
   def sendBeacon(url: String, data: BodyInit = js.native): Boolean = js.native
+
+  /** https://storage.spec.whatwg.org/ */
+  val storage: StorageManager = js.native
```
Comment (japgolly) [inline src/main/scala/org/scalajs/dom/lib.scala:330]: I had bad experience before with direct `val`s like this in facades when they aren't available in the target environment. Can you just please add a quick test to `SharedTests` to confirm that it works as expected in all environments, and if it isn't available in `JSDOMNode` can you pls make sure that another method of Navigator works as expected?

**Review by japgolly — CHANGES_REQUESTED** [review]

File: src/main/scala/org/scalajs/dom/experimental/storage/package.scala (hunk 1 above)
Comment (armanbilge) [reply src/main/scala/org/scalajs/dom/experimental/storage/package.scala:10]: Yes, you're right, this is for compat. I think I need to fix this in a couple other places as well.

File: src/main/scala/org/scalajs/dom/storage.scala (hunk 2 above)
Comment (armanbilge) [reply src/main/scala/org/scalajs/dom/storage.scala:13]: I think I need to fix this in several places too, sorry. So in general I should prefer `js.Object` to `js.Any`?

File: src/main/scala/org/scalajs/dom/lib.scala (hunk 3 above)
Comment (armanbilge) [reply src/main/scala/org/scalajs/dom/lib.scala:330]: Hmm, maybe safer to just make a `def`? That's what we decided in the contributing guidelines. This caught my eye too, but I erred on the side of not changing the code I was migrating.

File: src/main/scala/org/scalajs/dom/experimental/storage/package.scala (hunk 1 above)
Comment (armanbilge) [reply src/main/scala/org/scalajs/dom/experimental/storage/package.scala:10]:
```suggestion
  def toNavigatorStorage(navigator: dom.Navigator): dom.Navigator =
```

File: src/main/scala/org/scalajs/dom/storage.scala (hunk 2 above)
Comment (armanbilge) [reply src/main/scala/org/scalajs/dom/storage.scala]: Ahh, silly me, I didn't write this facade, just migrated it 😅 I'll try and fix this when I encounter it.

Comment (japgolly) [reply src/main/scala/org/scalajs/dom/storage.scala]: Yeah, with the facades have fields it should be `js.Object`. Normally I only extend `js.Any` for new enum types like `MediaStreamTrackState`

File: src/main/scala/org/scalajs/dom/lib.scala (hunk 3 above)
Comment (japgolly) [reply src/main/scala/org/scalajs/dom/lib.scala]: Yeah that's fine too

File: src/main/scala/org/scalajs/dom/storage.scala (hunk 2 above)
Comment (armanbilge) [reply src/main/scala/org/scalajs/dom/storage.scala]: That's an easy rule of thumb, thanks!

**Review by japgolly — APPROVED** [review]
Awesome! LGTM

**Pull request merged**